src/components/navbar/Navbar.tsx

```diff
@@ -1,30 +1,194 @@
+import Navbar from "react-bootstrap/Navbar";
+import Nav from "react-bootstrap/Nav";
+import Form from "react-bootstrap/Form";
+import FormControl from "react-bootstrap/FormControl";
+import Button from "react-bootstrap/Button";
+import InputGroup from "react-bootstrap/InputGroup";
+import { Dropdown } from "semantic-ui-react";
+
 import { useAuth0 } from "../../utils/react-auth0-spa";
-import { Link } from "react-router-dom";
+import { ProjectsContext } from "../../Store";
+import { api } from "../../api";
+import { FETCH_PROJECTS } from "../../utils/Types";
+import "./Navbar.css";
 
-const Navbar = (): JSX.Element => {
-  const { isAuthenticated, loginWithRedirect, logout } = useAuth0();
+const NavBar = (): JSX.Element => {
+  const { projectsState, projectsDispatch } = React.useContext(ProjectsContext);
+  const { getTokenSilently, user, isAuthenticated } = useAuth0();
 
-  console.log(loginWithRedirect);
+  const fetchProjects = useCallback(async () => {
+    const token = await getTokenSilently();
+    try {
+      const response = await api.get("projects", {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      });
+      return projectsDispatch({
+        type: FETCH_PROJECTS,
+        payload: response.data,
+      });
+    } catch (error) {
+      console.error(error);
+    }
+  }, [getTokenSilently, projectsDispatch]);
 
-  return (
-    <div>
-      {!isAuthenticated && (
-        <button onClick={(): any => loginWithRedirect({})}>Log in</button>
-      )}
-
-      {isAuthenticated && (
-        <button onClick={(): any => logout()}>Log out</button>
-      )}
-
-      {isAuthenticated && (
-        <span>
-          <Link to="/">Home</Link>&nbsp;
-          <Link to="/profile">Profile</Link>
+  useEffect(() => {
+    fetchProjects();
+  }, [fetchProjects]);
+
+  const renderIcon = (): JSX.Element => {
+    return (
+      <Navbar.Brand>
+        <span className="navbar-icon">
+          <i className="fab fa-reddit" />
         </span>
-      )}
-    </div>
+      </Navbar.Brand>
+    );
+  };
+
+  const renderSearch = (): JSX.Element => {
+    return (
+      <Form inline className="navbar-search-form">
+        <InputGroup className="mb-3 navbar-search-input">
+          <FormControl placeholder="Search" />
+          <InputGroup.Append>
+            <Button variant="outline-secondary">
+              <i className="fas fa-search" />
+            </Button>
+          </InputGroup.Append>
+        </InputGroup>
+      </Form>
+    );
+  };
+
+  const renderProjectsDropdown = (): JSX.Element => {
+    return (
+      <Dropdown text="Projects" className="navbar-projects-dropdown" inline>
+        <Dropdown.Menu className="navbar-projects-dropdown-menu">
+          {isAuthenticated ? (
+            Object.keys(projectsState).length === 0 ? (
+              ""
+            ) : (
+              Object.keys(projectsState)
+                .slice(0, 5)
+                .map((project: any) => {
```

Comment:
Replace any with something else.

Reply:
I don't think I can because projectsState is of type ProjectsInitialState.